docs/concepts/code-generation.md

```diff
@@ -28,13 +28,11 @@ export const api = generatedApi.enhanceEndpoints({
   endpoints: {
     // basic notation: just specify properties to be overridden
     getPetById: {
-      provides: (response) => [{ type: 'Pet', id: response.id }],
+      provides: (result) => (result ? [{ type: 'Pet', id: result.id }] : []),
     },
     findPetsByStatus: {
-      provides: (response) => [
-        { type: 'Pet', id: 'LIST' },
-        ...response.map((pet) => ({ type: 'Pet' as const, id: pet.id })),
-      ],
+      provides: (result) =>
+        result ? [{ type: 'Pet', id: 'LIST' }, ...result.map((pet) => ({ type: 'Pet' as const, id: pet.id }))] : [],
```

Comment:
```suggestion
        // is result available?
        result ? 
          // successful query
          [{ type: 'Pet', id: 'LIST' }, ...result.map((pet) => ({ type: 'Pet' as const, id: pet.id }))] : 
          // an error occured, but we still want to refetch this query when `{ type: 'Pet', id: 'LIST' }` is invalidated
          [{ type: 'Pet', id: 'LIST' }],
```

This is actually a perfect example why we did this - even in an error case we want the list to later be refetched.